projects/plugins/jetpack/extensions/blocks/payment-buttons/edit.js

```diff
@@ -65,18 +65,16 @@ function PaymentButtonsEdit( { clientId } ) {
 	const showStripeConnectAction = ! shouldUpgrade && ! isApiConnected && !! connectUrl;
 
 	const blockProps = useBlockProps();
-	const innerBlocksProps = useInnerBlocksProps(
-		{},
-		{
-			allowedBlocks: ALLOWED_BLOCKS,
-			orientation: 'horizontal',
-			template: [ [ 'jetpack/recurring-payments' ] ],
-			templateInsertUpdatesSelection: true,
-		}
-	);
+	const innerBlocksProps = useInnerBlocksProps( blockProps, {
+		allowedBlocks: ALLOWED_BLOCKS,
+		orientation: 'horizontal',
+		template: [ [ 'jetpack/recurring-payments' ] ],
+		templateInsertUpdatesSelection: true,
+		__experimentalLayout: layout,
+	} );
 
 	return (
-		<div { ...blockProps }>
+		<>
```

Comment:
This makes the Stripe nudge and the Jetpack upgrade nudge to be out of the block edit component, causing an overlap between the nudges and the block toolbar:

<img width="700" alt="Screen Shot 2022-09-15 at 13 43 09" src="https://user-images.githubusercontent.com/1233880/190395131-1bb2310d-badb-43d9-bf1c-121adbe121ac.png">

<img width="913" alt="Screen Shot 2022-09-15 at 13 44 37" src="https://user-images.githubusercontent.com/1233880/190395335-46b7d3b8-50d4-444b-8d1e-13cd5ce6f3d4.png">

Reply:
Hmm, the reason I made this change was because the flex container would then contain the div that wraps all of the buttons, rather than the buttons directly, setting a width on the buttons then sets the wrong width, see below. With the nudge it wraps both the nudge and the button container.

![image](https://user-images.githubusercontent.com/93301/190422509-28b11567-d8c1-4742-a4e6-164de0b436e8.png)

Seems that with the nudge, this approach isn't viable 👎 I'll have to think of something else.

Reply:
Oooh, I see – that's unfortunate.

> I'll have to think of something else.

If that proves to be complex (something that I kinda expect), we have the alternative approach of moving the nudges to the block edit component after it has been rendered, which is how we add the upgrade nudges to the paid blocks in WP.com:

https://github.com/Automattic/jetpack/blob/28ca04ee102e28945c99ecaed3a8122d4709dde5/projects/plugins/jetpack/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx#L74-L100

Reply:
Oh of course the wpcom and jp implementations of upgrade nudges are completely separate 😬

Reply:
For wpcom this is broken too albeit in a slightly different way, for basically the same reason - the outer div had a display:flex rule thanks to the is-layout-grid class.  ![image](https://user-images.githubusercontent.com/93301/190619014-24553b3d-81de-438f-8f58-a7398ff1652b.png)

The menu ultimately works by looking for the div with the clientId, so we have to put the upgrade nudges inside that div.

The layouts work [at edit time](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/hooks/layout.js#L384-L388) by applying the `is-layout-flex` css class to the `BlockListBlock` (i.e. the component rendered on canvas) and [at view time](https://github.com/WordPress/gutenberg/blob/trunk/lib/block-supports/layout.php#L383-L390) by adding the same class name to the outermost render content.

I think the solution is to change the edit to go back to `<div { ...blockProps }>` but to delete blockProps.className first. This way the display:flex won't be applied in the wrong place. I'll give this a try.

Reply:
Seems to work fine 👍

Reply:
> For wpcom this is broken too albeit in a slightly different way, for basically the same reason - the outer div had a display:flex rule thanks to the is-layout-grid class.

I still reproduce the same, not sure if I'm missing something but I made sure that my sandbox is using the most recent changes

<img width="674" alt="Screen Shot 2022-09-16 at 16 32 14" src="https://user-images.githubusercontent.com/1233880/190663678-89bb426c-2d26-4bfa-a72b-ec36d52ef58e.png">

Reply:
You're absolutely right, it's still broken on wpcom. I don't really understand why, for some reason the is-layout-flex (and other classes) are still being applied to the outer div on wpcom but not on jp.

Also both the outer and list blocks div get the same id, so if trying to make this approach work, that needs addressing too.

Reply:
This is basically the same problem as we had before in #25102 - the wpcom upgrade nudge hooks itself inside the edit component and we can't shift it elsewhere as we're not in control of it.

I don't think this can fundamentally work without first changing how the upgrade nudge works. 😞

Reply:
I've addressed this issue by creating a div for the wpcom upgrade nudge to be placed in, rather than allowing the wpcom code to create the nudge div where it thinks is best.

It works, I _think_ it's ok but please let me know if this is too much of a hack. Ideally I think the wpcom nudge would work rather differently.